src/libertem/io/dataset/mib.py

```diff
@@ -444,28 +481,44 @@ class MIBDataSet(DataSet):
     path: str
         Path to either the .hdr file or one of the .mib files
 
-    scan_size: tuple of int, optional
-        A tuple (y, x) that specifies the size of the scanned region. It is
-        automatically read from the .hdr file if you specify one as `path`.
+    nav_shape: tuple of int, optional
+        A n-tuple that specifies the size of the navigation region ((y, x), but
+        can also be of length 1 for example for a line scan, or length 3 for
+        a data cube, for example)
+
+    sig_shape: tuple of int, optional
+        Common case: (height, width); but can be any dimensionality
+
+    sync_offset: int, optional
+        If positive, number of frames to skip from start
+        If negative, number of blank frames to insert at start
     """
-    def __init__(self, path, tileshape=None, scan_size=None, disable_glob=False):
+    def __init__(self, path, tileshape=None, scan_size=None, disable_glob=False,
+                 nav_shape=None, sig_shape=None, sync_offset=0):
         super().__init__()
         self._sig_dims = 2
         self._path = path
+        self._nav_shape = tuple(nav_shape) if nav_shape else nav_shape
+        self._sig_shape = tuple(sig_shape) if sig_shape else sig_shape
+        self._sync_offset = sync_offset
         # handle backwards-compatability:
         if tileshape is not None:
             warnings.warn(
                 "tileshape argument is ignored and will be removed after 0.6.0",
                 FutureWarning
             )
         if scan_size is not None:
-            scan_size = tuple(scan_size)
-        else:
-            if not path.lower().endswith(".hdr"):
-                raise ValueError(
-                    "either scan_size needs to be passed, or path needs to point to a .hdr file"
+            warnings.warn(
+                "scan_size argument is ignored and will be removed after 0.6.0",
```

Comment:
The warning text can be improved, I think: 1) the argument is not really ignored, 2) the user should be pointed to the new argument name `nav_shape` directly in the message. Other than that the deprecation looks good!